openreview/conference/templates/areachairWebfield.js

```diff
@@ -651,12 +672,94 @@ var registerEventHandlers = function() {
   });
 
   $('#group-container').on('click', 'a.collapse-btn', function(e) {
-    $(this).next().slideToggle();
+    // $(this).next().slideToggle();
     if ($(this).text() === 'Show reviewers') {
       $(this).text('Hide reviewers');
     } else {
       $(this).text('Show reviewers');
     }
+    // return false;
+  });
+
+  $('#group-container').on('click', 'button.btn.assign-reviewer-button', function(e) {
+    var $link = $(this);
+    var paperNumber = $link.data('paperNumber');
+    var paperForum = $link.data('paperForum');
+    console.log(paperNumber);
+    console.log(paperForum);
+    var $currDiv = $('#' + paperForum + '-add-reviewer');
+    console.log($currDiv);
+    userToAdd = $currDiv.find('input').val().trim();
+    console.log(userToAdd);
+    nextAnonNumber = findNextAnonGroupNumber(paperNumber);
+    console.log(nextAnonNumber);
+    $.when(
+      Webfield.put('/groups/members', {
+        id: CONFERENCE_ID + '/Paper' + paperNumber + '/Reviewers',
+        members: [userToAdd]
+      }),
+      Webfield.post('/groups', {
+        id: CONFERENCE_ID + '/Paper' + paperNumber + '/AnonReviewer' + nextAnonNumber,
+        members: [userToAdd],
+        readers: [
+          CONFERENCE_ID + '/Program_Chairs',
+          CONFERENCE_ID + '/Paper' + paperNumber + '/Area_Chairs',
+          CONFERENCE_ID + '/Paper' + paperNumber + '/AnonReviewer' + nextAnonNumber],
+        writers: [
+          CONFERENCE_ID + '/Program_Chairs',
+          CONFERENCE_ID + '/Paper' + paperNumber + '/Area_Chairs'],
+        signatures: [CONFERENCE_ID + '/Paper' + paperNumber + '/Area_Chairs'],
+        signatories: [CONFERENCE_ID + '/Paper' + paperNumber + '/AnonReviewer' + nextAnonNumber]
+      }),
+      promptMessage('Reviewer ' + view.prettyId(userToAdd) + ' has been assigned to paper ' + paperNumber)
```

Comment:
you need to wait until both promises are finished to prompt the message, also we need to catch the errors. For example:

```
$.when(
   Webfield.post(...),
   Webfield.post(...)
)
.then(function(results) {
   promptMessage
})
```

I think the errors are being catch in the Webfield.js code but I'm not sure.

Reply:
The errors are being caught by the Webfield.post function. It will automatically show the error prompt if one of the requests fails

Reply:
But yes I agree with Melisa, the promptMessage should be in the `.then` callback

Reply:
Agreed. This has been corrected.